specs/fulu/p2p-interface.md

```diff
@@ -292,6 +292,111 @@ gossip. In particular, clients MUST:
 - Update gossip rule related data structures (i.e. update the anti-equivocation
   cache).
 
+#### Partial Columns
+
+Gossipsub's [Partial Message
+Extension](https://github.com/libp2p/specs/pull/685) enables exchanging
+selective parts of a message rather than the whole. The specification here
+describes how Consensus Clients use Partial Messages to disseminate cells.
+
+*Editor Note*: This change MUST NOT be merged before the linked libp2p/specs.
+
+##### `PartialDataColumnSidecar`
+
+The `PartialDataColumnSidecar `is identical to the `DataColumnSidecar `container
+with the exception that fields and cells may be None.
+
+Some fields are only set when a peer is eagerly pushing this container.
+Otherwise, if a peer is sending the container as a response, there is no need to
+include redundant fields.
+
+```python
+class PartialDataColumnSidecar(Container):
+    signed_block_header: SignedBeaconBlockHeader
+    cells_present_bitmap: Bitlist[MAX_BLOB_COMMITMENTS_PER_BLOCK]
+    column: List[Cell, MAX_BLOB_COMMITMENTS_PER_BLOCK]
+    kzg_proofs: List[KZGProof, MAX_BLOB_COMMITMENTS_PER_BLOCK]
+```
+
+##### Partial IHAVE/IWANT
+
+Peers communicate IHAVE and IWANTS succintly with a bitmap. For IHAVE, a set bit (`1`)
+means that the peer has the corresponding cell. For IWANT, a set bit means the
+peer wants the corresponding cell. The bitmap is encoded by the following function:
+
+```python
+def create_bitmap(set_indices, n):
+    bitmap = [0] * ((n + 7) // 8)
+    for i in set_indices:
+        bitmap[i // 8] |= 1 << (i % 8)
+    return bitmap
+```
+
+For example, if the peer wanted cells at positions 0,3,9, the corresponding
+IWANT bitmap would be: `0000 1001 000 0010`.
```

Comment:
What's `n` in the example?
000 -> 0000 , and also the byte order looks wrong, shouldn't we have MSB-first?

Reply:
thanks for the 000 -> 0000 correction.

n can be anything from 9 to 16 in this case, but I should just say 12 to be clear in the example.

> The byte order looks wrong, shouldn't we have MSB-first?

https://www.online-python.com/ietUIq50Ol